pycloudlib/lxd/instance.py

```diff
@@ -333,3 +334,85 @@ def wait_for_stop(self):
                 return
             time.sleep(1)
         raise TimeoutError
+
+
+class LXDVirtualMachineInstance(LXDInstance):
+    """LXD Virtual Machine backed instance."""
+
+    def __init__(self, name, key_pair=None, execute_via_ssh=True):
+        """Set up instance.
+
+        Args:
+            name: name of instance
+            key_pair: SSH key object
+        """
+        super().__init__(
+            name=name,
+            key_pair=key_pair,
+            execute_via_ssh=execute_via_ssh
+        )
+
+        self.series = self._extract_release_from_instance_info()
+
+    def _run_command(self, command, stdin):
+        """Run command in the instance."""
+        if self.execute_via_ssh:
+            return super()._run_command(command, stdin)
+
+        if self.series == "xenial":
+            msg = (
+                "The xenial instance does not support executing commads"
+                " with exec."
+            )
+            raise Exception(msg)
+
+        base_cmd = [
+            'lxc', 'exec', self.name, '--', 'sudo', '-u', self.username, '--'
+        ]
+        return subp(base_cmd + list(command), rcs=None)
```

Comment:
No need to repeat this implementation here, I think. We can probably do something like:

```py
if self.series == "xenial" and not self.execute_via_ssh:
   # raise/log here
super()._run_command(command, stdin)
```

That way we only have the actual `_run_command` code defined in a single place.

Reply:
I don't know If I am missing something here, but I think this will not hold if we are using `xenial containers` which support both operations.

Reply:
This implementation is in `LXDVirtualMachineInstance`, it shouldn't be called in the container situation.

Reply:
Ohhh okay, now I see your point. I thought you meant adding that part to the `_run_command`  we have at `LXDInstance`. The proposed change makes sense. I will update the code